net/server.go

```diff
@@ -198,24 +198,31 @@ func (s *server) GetRecords(ctx context.Context, req *pb.GetRecordsRequest) (*pb
 
 	for _, lg := range info.Logs {
 		var (
-			offset cid.Cid
-			limit  int
-			pblg   *pb.Log
+			offset  cid.Cid
+			limit   int
+			counter int64
+			pblg    *pb.Log
 		)
 		if opts, ok := reqd[lg.ID]; ok {
 			offset = opts.Offset.Cid
+			counter = opts.Counter
 			limit = minInt(int(opts.Limit), logRecordLimit)
 		} else {
 			offset = cid.Undef
 			limit = logRecordLimit
-			pblg = logToProto(lg)
+			counter = thread.CounterUndef
 		}
+		pblg = logToProto(lg)
```

Comment:
uber nit: since it's unconditionally initialized, it can be put into the `var()` section above, or to save some CPU cycles, don't create the var until L245, and get the counter from `lg.Head.Counter` in L221.